components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -5,3 +5,597 @@
+
+	-- get participants data for the map + get map mode
+	map = CustomMatchGroupInput._processPlayerMapData(map, match)
+
+	-- set sumscore to 0 if it isn't a number
+	if String.isEmpty(match.opponent1.sumscore) then
+		match.opponent1.sumscore = 0
+	end
+	if String.isEmpty(match.opponent2.sumscore) then
+		match.opponent2.sumscore = 0
+	end
+
+	--adjust sumscore for winner opponent
+	if (tonumber(map.winner) or 0) > 0 then
+		match['opponent' .. map.winner].sumscore =
+			match['opponent' .. map.winner].sumscore + 1
+	end
+
+	match['map' .. mapIndex] = map
+
+	return match, subGroupIndex
+end
+
+function CustomMatchGroupInput._mapWinnerProcessing(map)
+	map.scores = {}
+	local hasManualScores = false
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if Logic.isNotEmpty(score) then
+			hasManualScores = true
+			score = CONVERT_STATUS_INPUT[score] or score
+			if Logic.isNumeric(score) then
+				obj.status = 'S'
+				obj.score = score
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+
+	if hasManualScores then
+		for scoreIndex, _ in Table.iter.spairs(indexedScores, CustomMatchGroupInput._placementSortFunction) do
+			if not tonumber(map.winner or '') then
+				map.winner = scoreIndex
+			else
+				break
+			end
+		end
+	else
+		local winnerInput = tonumber(map.winner)
+		if String.isNotEmpty(map.walkover) then
+			local walkoverInput = tonumber(map.walkover)
+			if walkoverInput == 1 then
+				map.winner = 1
+			elseif walkoverInput == 2 then
+				map.winner = 2
+			elseif walkoverInput == 0 then
+				map.winner = 0
+			end
+			map.walkover = Table.includes(ALLOWED_STATUSES, map.walkover) and map.walkover or 'L'
+			map.scores = {-1, -1}
+			map.resulttype = 'default'
+		elseif map.winner == 'skip' then
+			map.scores = {0, 0}
+			map.scores = {-1, -1}
+			map.resulttype = 'np'
+		elseif winnerInput == 1 then
+			map.scores = {1, 0}
+		elseif winnerInput == 2 then
+			map.scores = {0, 1}
+		end
+	end
+
+	return map
+end
+
+function CustomMatchGroupInput._processPlayerMapData(map, match)
+	local participants = {}
+	local submatchOpponentPlayerNumbers = {}
+
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local opponent = match['opponent' .. opponentIndex]
+
+		if opponent.type == Opponent.team then
+			error('Team matches not yet supported')
+		elseif opponent.type == Opponent.literal then
+			table.insert(submatchOpponentPlayerNumbers, opponent.type)
+		else
+			table.insert(submatchOpponentPlayerNumbers, #opponent.match2players)
+
+			CustomMatchGroupInput._processDefaultPlayerMapData(
+				opponent.match2players or {},
+				opponentIndex,
+				map,
+				participants
+			)
+		end
+	end
+
+	map.mode = table.concat(submatchOpponentPlayerNumbers, 'v')
```

Comment:
```suggestion
	for opponentIndex = 1, MAX_NUM_OPPONENTS do
		local opponent = match['opponent' .. opponentIndex]

		if Opponent.typeIsParty(opponent.type) then
			CustomMatchGroupInput._processDefaultPlayerMapData(
				opponent.match2players or {},
				opponentIndex,
				map,
				participants
			)
		end
	end

	map.mode = Opponent.toMode(match.opponent1.type, match.opponent2.type)
```

Reply:
```suggestion
	for opponentIndex = 1, MAX_NUM_OPPONENTS do
		local opponent = match['opponent' .. opponentIndex]
		if Opponent.typeIsParty(opponent.type) then
			CustomMatchGroupInput._processDefaultPlayerMapData(
				opponent.match2players or {},
				opponentIndex,
				map,
				participants
			)
		elseif opponent.type == Opponent.team then
			error('Team matches not yet supported')
		end
	end

	map.mode = Opponent.toMode(match.opponent1.type, match.opponent2.type)
```

want the error there for easier finding where to copy paste stuff when team/submatch support is added

Reply:
Any perticual reason this was undid in its full?

Reply:
copy paste mistake